hal/src/nRF52840/i2c_hal.cpp

```diff
@@ -310,28 +313,46 @@ int hal_i2c_init(hal_i2c_interface_t i2c, const hal_i2c_config_t* config) {
     os_thread_scheduling(false, nullptr);
     if (i2cMap[i2c].mutex == nullptr) {
         os_mutex_recursive_create(&i2cMap[i2c].mutex);
-    } else {
-        // Already initialized
-        os_thread_scheduling(true, nullptr);
-        return SYSTEM_ERROR_NONE;
-    }
+    } 
 
     // Capture the mutex and re-enable threading
     I2cLock lk(i2c);
     os_thread_scheduling(true, nullptr);
 
-    // Initialize internal data structure
+    if (i2cMap[i2c].configured) {
+        // Configured, but new buffers are invalid
+        if (!isConfigValid(config)){
```

Comment:
Nit: Can probably do this just once and short-circuit early on before configuring/reconfiguring.

Reply:
The previous logic allowed initializing the HAL with a NULL config (ie it should use default buffers). Do we want to disallow this?

Reply:
Should still be allowed, yeah, let's leave as-is :)